src/libsync/propagateupload.cpp

```diff
@@ -211,12 +211,51 @@ void PropagateUploadFileCommon::setDeleteExisting(bool enabled)
     _deleteExisting = enabled;
 }
 
+bool PropagateUploadFileCommon::correctInvalidFileName()
+{
+    const auto originalFilePathSyncFolder = _item->_file;
+    auto slashPosition = originalFilePathSyncFolder.lastIndexOf('/');
+    const auto originalFileParentPathSyncFolder = slashPosition >= 0 ? originalFilePathSyncFolder.left(slashPosition) : QString();
+    const auto newFileName = originalFilePathSyncFolder.mid(slashPosition + 1).trimmed();
+    const auto originalFilePathAbsolute = propagator()->fullLocalPath(_item->_file);
+    slashPosition = originalFilePathAbsolute.lastIndexOf('/');
+    const auto orignalFileParentPathAbsolute = originalFilePathAbsolute.left(slashPosition);
+    const QString newFilePathAbsolute = orignalFileParentPathAbsolute + QStringLiteral("/") + newFileName;
+
+    if (newFilePathAbsolute != originalFilePathAbsolute) {
+        QFileInfo newFileInfo(newFilePathAbsolute);
+        if (newFileInfo.exists()) {
+            done(SyncFileItem::NormalError, tr("File contains trailing spaces and coudn't be renamed, because a file with the same name already exists."));
+            return false;
+        }
+
+        QFile originalFile(originalFilePathAbsolute);
+        const auto renameSuccess = originalFile.rename(newFilePathAbsolute);
```

Comment:
what is in the sync database after we execute this `rename` ?

Reply:
Nothing. At this point nothing has been written to the sync database.